### ISSUE TITLE
Color Picker: text input in nested dialogue

### ISSUE THREAD

**Issue by bda94** [primefaces/primefaces#11076]
Hello, 

I noticed in my application and even in the primefaces showcase that the color picker, if used inside a dialog, has a non-functioning input field.

![image](https://github.com/primefaces/primefaces/assets/152994368/74454a98-2bc4-4307-8430-52acf5ca488d)

I can copy and paste value to the color field, but I cannot manually write or delete values. I do not think this is the intended behaviour since in the other color pickers in the showcase it is possible to do so.

### JSF version

2.3.9

### Java version

21

**Comment by melloware** [comment]
I will take a look.  I thought I had tested it but never realized the input field was non editable.

**Comment by melloware** [comment]
Fixed for 13.0.4.  Here is a MonkeyPatch if you need it

```js
if (PrimeFaces.widget.ColorPicker) {
    PrimeFaces.widget.ColorPicker.prototype.bindInputCallbacks = function() {
        var $this = this;
        if ($this.hasBehavior('change')) {
            $this.input.on('coloris:pick', function(e) {
                $this.callBehavior('change');
            });
        }
        if ($this.hasBehavior('open') || $this.cfg.parent) {
            $this.input.on('open.colorpicker', function(e) {
                $this.callBehavior('open');

                if ($this.cfg.parent) {
                    // #11076 dialog support of input
                    var dialog = $($this.cfg.parent);
                    var colorInput = dialog.find('#clr-color-value');
                    colorInput.zIndex(9999);
                    colorInput.parent().zIndex(9999);
                }
            });
        }

        $this.input.on('close.colorpicker', function(e) {
            if ($this.hasBehavior('close')) {
                $this.callBehavior('close');
            }
            if ($this.hasFloatLabel) {
                var container = $this.input.parent();
                setTimeout(function() {
                    container.removeClass('ui-inputwrapper-focus');
                    PrimeFaces.utils.updateFloatLabel(container, $this.input, $this.hasFloatLabel);
                }, 0);
            }
        });

        if ($this.hasFloatLabel) {
            $this.input.on('focus.colorpicker', function() {
                $this.input.parent().addClass('ui-inputwrapper-focus');
            });
        }
    }
};
```
